testScripts/createAKSFile.md

```diff
@@ -0,0 +1,214 @@
+This should create a file name azure-vote.yaml
+
+The ultimate fix to innovation engine would be to remove all leading spaces from new lines so 
+that this is not an issue. This issue persists as even if it completes the creation of the YAML
+file the YAML file is tabbed and thus
+
+Gives this error "error: error parsing azure-vote-base.yaml: error converting YAML to JSON: yaml: 
+line 34: could not find expected ':'"
+
+```azurecli-interactive
+export AKS_FILE_NAME=azure-vote-test2.yaml
+```
+
+        ```azurecli-interactive
+        cat <<EOF > $AKS_FILE_NAME
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+        name: azure-vote-back
+        spec:
+        replicas: 1
+        selector:
+            matchLabels:
+            app: azure-vote-back
+        template:
+            metadata:
+            labels:
+                app: azure-vote-back
+            spec:
+            nodeSelector:
+                "kubernetes.io/os": linux
+            containers:
+            - name: azure-vote-back
+                image: mcr.microsoft.com/oss/bitnami/redis:6.0.8
+                env:
+                - name: ALLOW_EMPTY_PASSWORD
+                value: "yes"
+                resources:
+                requests:
+                    cpu: 100m
+                    memory: 128Mi
+                limits:
+                    cpu: 250m
+                    memory: 256Mi
+                ports:
+                - containerPort: 6379
+                name: redis
+        ---
+        apiVersion: v1
+        kind: Service
+        metadata:
+        name: azure-vote-back
+        spec:
+        ports:
+        - port: 6379
+        selector:
+            app: azure-vote-back
+        ---
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+        name: azure-vote-front
+        spec:
+        replicas: 1
+        selector:
+            matchLabels:
+            app: azure-vote-front
+        template:
+            metadata:
+            labels:
+                app: azure-vote-front
+            spec:
+            nodeSelector:
+                "kubernetes.io/os": linux
+            containers:
+            - name: azure-vote-front
+                image: mcr.microsoft.com/azuredocs/azure-vote-front:v1
+                resources:
+                requests:
+                    cpu: 100m
+                    memory: 128Mi
+                limits:
+                    cpu: 250m
+                    memory: 256Mi
+                ports:
+                - containerPort: 80
+                env:
+                - name: REDIS
+                value: "azure-vote-back"
+        ---
+        apiVersion: v1
+        kind: Service
+        metadata:
+        name: azure-vote-front
+        spec:
+        type: LoadBalancer
+        ports:
+        - port: 80
+        selector:
+            app: azure-vote-front
+
+        EOF
+        ```
+
+```azurecli-interactive
+echo "hello world"
```

Comment:
Why?

I'm stopping the review here, I'm not sure what the point of this file is and the beginning says it is broken anyway.

Tests need to be readable and meanigful.  If the goal is to demonstrate the break then it should be in a working branch for that break. Create an issue, create a branch commit the test to that branch, fix it, merge that branch.

Reply:
We can remove this line. The final echo was part of the testing process. The beginning of this test was meant to explain the test. There was an issue with indentation when creating YAML files. I added the note at the top to say that this is no longer an issue, but I wanted to leave the description of the issue solved to show the history. I can do this easily by just rewriting the initial description.